File: Hash Table/3217. Delete Nodes From Linked List Present in Array.py
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        ans = ListNode(0)
        tail = ans
        blocked_nums = set(nums)
        while head:
            if head.val not in blocked_nums:
                tail.next = ListNode(head.val)
                tail = tail.next
            head = head.next
        return ans.next

File: Hash Table/3217. Delete Nodes From Linked List Present in Array.cpp
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) 
    {
        set<int>st;
        for(auto num:nums) st.insert(num);
        
        ListNode* ans = new ListNode(0);
        ListNode* tail = ans;
        while(head)
        {
            int val = head->val; head = head->next;
            if(not st.count(val))
            {
                tail->next = new ListNode(val);
                tail = tail->next;
            }
        }
        return ans->next;
    }
};